ui/src/tool-panel/tool-panel.tsx

```diff
@@ -254,28 +268,31 @@ const ToolPanel: React.FC<ToolPanelProps> = ({
             } as ToolPanelTab;
         });
 
-        const geometryKmPostTabs = geometryKmPosts.map((k) => {
-            return {
-                asset: { type: 'GEOMETRY_KM_POST', id: k.geometryItem.id },
-                title: k.geometryItem.kmNumber,
-                element: (
-                    <GeometryKmPostInfobox
-                        geometryKmPost={k.geometryItem}
-                        planId={k.planId}
-                        onShowOnMap={() =>
-                            k.geometryItem.location &&
-                            showArea(
-                                calculateBoundingBoxToShowAroundLocation(k.geometryItem.location),
-                            )
-                        }
-                        visibilities={infoboxVisibilities.geometryKmPost}
-                        onVisibilityChange={(visibilities: GeometryKmPostInfoboxVisibilities) =>
-                            infoboxVisibilityChange('geometryKmPost', visibilities)
-                        }
-                    />
-                ),
-            } as ToolPanelTab;
-        });
+        const geometryKmPostTabs = geometryKmPostIds.map(
+            (k: SelectedGeometryItemId<LayoutKmPostId>) => {
+                const kmPost = getPlan(k.planId)?.kmPosts?.find((p) => p.id === k.id);
```

Comment:
Mietin että olisikohan tässä mahdollista käyttää jotain massahakua jo tämän `.map`:in ulkopuolella. Sen lisäksi että saataisiin haettua kaikki kerralla, niin sen voisi wrapata `useLoaderWithStatus`:iin, jolloin voitaisiin haun failatessa näyttää jotain muuta kuin ikuista spinneriä

Reply:
Joo se voisi olla fiksu.

Backari ei nykyisellään tue massa-hakua mutta tämän nykyinen käyttötapa on tämän komponentin kannalta ikäänkuin se olisi jo massa-haku. Se poimii tuossa ylempänä kaikki ne planit joita tulee tarvitsemaan (+unique filter) ja hakee ne kerralla. Tämä get-funkkari ei tee mitään kutsua vaan poimii vain esi-haetuista suunnitelmista id:llä yhden ja kaivaa sieltä tuon kmPostin.

Sen voisi kyllä pistää loaderiin yhtä hyvin - tuo nykyrakenne on vähän hassu mutta laitoin sen samaan mukaan kun kaikki muukin async haetaan täällä noin. Voisin kurkata olisiko se loaderilla nätimpi. Siitä voisi tosiaan ainakin virheilmon vaihtaa spinnerin tilalle samalla, mutta vaikka tuo paneeli jää spinneriksi, se ei minusta ole iso ongelma. Failannut kutsuhan ilmoittaa joka tapauksessa toasterilla käyttäjälle että kutsu failasi. Tuo ilmoitus + valkoiseksi jäävä komponentti kertoo minusta jo ihan kohtuullisesti mitä tapahtui.

Reply:
Päätin jättää tämän toistaiseksi tekemättä. Tuo nykyrakenne käyttää useLoaderia mutta ilman statusta, mutta nähdäkseni syy miksi se on tehty noin on että komponentti hakee kaiken datan ensin ja vasta sitten piirtää jotain uutta. Eli vähentää jotain välkyntää/tarpeetonta redraw:ta. En ole ihan varma kannattaako sitä tehdä noin mutta en nyt tässä yhteydessä lähde sitä purkamaankaan.